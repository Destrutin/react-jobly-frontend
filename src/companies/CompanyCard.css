.CompanyCard-card {
  background-color: #e6e6fa;
  border-radius: 10px;
  border: 1px solid #9370db;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
  transition: transform 0.3s ease;
  text-decoration: none;
  color: inherit;
  display: flex;
  flex-direction: column;
  width: 400px;
  margin: 0 auto;
}

.CompanyCard-card:hover {
  transform: translateY(-4px);
}

.CompanyCard-header {
  padding: 1rem;
  border-bottom: 1px solid #e0e0e0;
}

.CompanyCard-body {
  padding: 1rem;
}

.CompanyCard-logo {
  width: 100%;
  max-height: 100px;
  object-fit: contain;
  border-radius: 8px;
}

.CompanyCard-name {
  font-size: 1.2rem;
  font-weight: 600;
  color: #333333;
  margin-bottom: 0.5rem;
}

.CompanyCard-description {
  font-size: 1rem;
  color: #666666;
  margin-bottom: 1rem;
}

.CompanyCard-readMore {
  color: #007bff;
  text-decoration: none;
  font-size: 1rem;
  align-self: flex-end;
}

.CompanyCard-readMore:hover {
  text-decoration: underline;
}
