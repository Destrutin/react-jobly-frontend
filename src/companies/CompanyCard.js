import React from "react";
import {Link} from "react-router-dom";


function CompanyCard({handle, name, description, logoUrl}) {
    return (
        <Link className="CompanyCard-card" to={`/companies/${handle}`}>
            <div className="card-body">
                <h4>{name}</h4>
                <p>{description}</p>
                {logoUrl && <img src={logoUrl} alt={name} className="card-img"/>}
            </div>
        </Link>
    )
}

export default CompanyCard;