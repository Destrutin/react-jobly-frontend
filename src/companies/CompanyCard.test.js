import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom';
import CompanyCard from './CompanyCard';

test('renders company card without crashing', () => {
    render(
        <MemoryRouter>
            <CompanyCard
                handle="test"
                name="test"
                description="test"
                logoUrl="test.png"
            />
        </MemoryRouter>
    )
})

test('matches snapshot', () => {
    const {asFragment} = render(
        <MemoryRouter>
            <CompanyCard
            handle="test"
            name="test"
            description="test"
            logoUrl="test.png"
            />
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})