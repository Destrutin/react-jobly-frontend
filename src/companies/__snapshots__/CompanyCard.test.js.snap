// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard-card"
    href="/companies/test"
  >
    <div
      class="card-body"
    >
      <h4>
        test
      </h4>
      <p>
        test
      </p>
      <img
        alt="test"
        class="card-img"
        src="test.png"
      />
    </div>
  </a>
</DocumentFragment>
`;
